### PR TITLE
BREAKING CHANGE feat: add custom HTTP header support

### PR DESCRIPTION
**Pull request by eshanholtz**

### CODE REVIEW

File: tests/integration/authy/v1/service/test_entity.py [hunk 1]
```diff
@@ -19,10 +19,16 @@ def test_create_request(self):
...
+                                .entities.create(identity="identity", twilio_authy_sandbox_mode="twilio_authy_sandbox_mode")
 
         values = {'Identity': "identity", }
 
+        headers = {'Twilio-Authy-Sandbox-Mode': "twilio_authy_sandbox_mode", }
+        self.holodeck.assert_has_request(Request(
+            'post',
+            'https://authy.twilio.com/v1/Services/ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/Entities',
+            headers=headers,
+        ))
```
Comment (childish-sambino) [inline tests/integration/authy/v1/service/test_entity.py]: Why is a second assertion added?

File: twilio/rest/accounts/v1/credential/aws.py [hunk 2]
```diff
@@ -86,13 +86,9 @@ def page(self, page_token=values.unset, page_number=values.unset,
...
         """
-        params = values.of({'PageToken': page_token, 'Page': page_number, 'PageSize': page_size, })
+        data = values.of({'PageToken': page_token, 'Page': page_number, 'PageSize': page_size, })
 
-        response = self._version.page(
-            'GET',
-            self._uri,
-            params=params,
-        )
+        response = self._version.page(method='GET', uri=self._uri, params=data, )
```
Comment (childish-sambino) [inline twilio/rest/accounts/v1/credential/aws.py]: Why the name change?

File: twilio/rest/accounts/v1/credential/aws.py [hunk 3]
```diff
@@ -116,13 +112,13 @@ def get_page(self, target_url):
...
         """
-        Create a new AwsInstance
+        Create the AwsInstance
 
         :param unicode credentials: A string that contains the AWS access credentials in the format <AWS_ACCESS_KEY_ID>:<AWS_SECRET_ACCESS_KEY>
         :param unicode friendly_name: A string to describe the resource
         :param unicode account_sid: The Subaccount this Credential should be associated with.
 
-        :returns: Newly created AwsInstance
+        :returns: The created AwsInstance
```
Comment (childish-sambino) [inline twilio/rest/accounts/v1/credential/aws.py]: Was this intentional?

File: twilio/rest/accounts/v1/credential/aws.py [hunk 4]
```diff
@@ -272,7 +254,7 @@ def delete(self):
         :returns: True if delete succeeds, False otherwise
         :rtype: bool
         """
-        return self._version.delete('delete', self._uri)
+        return self._version.delete(method='DELETE', uri=self._uri, )
```
Comment (childish-sambino) [inline twilio/rest/accounts/v1/credential/aws.py]: Seeing a lot of changes not (directly) related to header support. Can these be split out?

File: twilio/rest/accounts/v1/credential/aws.py (hunk 3 above)
Comment (eshanholtz) [reply twilio/rest/accounts/v1/credential/aws.py]: Kind of... side effect of refactoring yoyodyne to reduce duplications.

File: tests/integration/authy/v1/service/test_entity.py (hunk 1 above)
Comment (eshanholtz) [reply tests/integration/authy/v1/service/test_entity.py]: Easier logic to add to the integration test generator than rewriting the existing assert to include headers. It matches the Node helper lib header integration tests

File: twilio/rest/accounts/v1/credential/aws.py (hunk 2 above)
Comment (eshanholtz) [reply twilio/rest/accounts/v1/credential/aws.py]: Mainly a side effect of refactoring. It's just an internal variable name, so it shouldn't matter.

File: twilio/rest/accounts/v1/credential/aws.py (hunk 4 above)
Comment (eshanholtz) [reply twilio/rest/accounts/v1/credential/aws.py]: They _can_ but I'm not they _should_. Lots of the changes come from refactoring lots of duplicates and very near duplicates in yoyodyne to simplify the process of generating action functions. If I were the split out the changes, I likely wouldn't pass the SonarQube check on the yoyo side of things.

Comment (childish-sambino) [reply twilio/rest/accounts/v1/credential/aws.py]: What about just excluding the `header_support` flag?

Comment (eshanholtz) [reply twilio/rest/accounts/v1/credential/aws.py]: Oh, you mean push the refactor first _then_ the header support. Yeah I can totally do that.

Comment (eshanholtz) [reply twilio/rest/accounts/v1/credential/aws.py]: https://github.com/twilio/twilio-python/pull/502

**Pull request merged**